src/net_processing.cpp

```diff
@@ -855,8 +855,8 @@ void PeerLogicValidation::BlockChecked(const CBlock& block, const CValidationSta
 
     int nDoS = 0;
     if (state.IsInvalid(nDoS)) {
-        if (it != mapBlockSource.end() && State(it->second.first)) {
-            assert (state.GetRejectCode() < REJECT_INTERNAL); // Blocks are never rejected with internal reject codes
```

Comment:
The if statement has been modified to check that `state.GetRejectCode() < REJECT_INTERNAL` (so this assert can now never fail)